test/testing_utils.py

```diff
@@ -25,12 +28,35 @@ def assert_dict_allclose(actual, desired, atol=DEFAULT_TOL, rtol=DEFAULT_TOL, pa
     assumed to be numpy arrays
 
     The keys are checked first to make sure that they match
+
+    Parameters
+    ----------
+    actual : 1D dictionary
+        A dictionary of str: array_like pairs
+    desired : 1D dictionary
+        A dictionary of str: array_like pairs
+    atol : float
+        Absolute tolerance
+    rtol : float
+        Relative tolerance
+    partial : bool
+        Whether the assertion allows for partial keys, i.e. when the dictionaries
+        do not have matching keys.
+        If True, we assume that ``desired`` contains a subset of the keys of ``actual``.
+
+    Raises
+    ------
+    AssertionError
+        if the values for any key do not match to the target tolerance.
     """
     if not partial:
-        assertEqual(set(actual.keys()), set(desired.keys()))
-    for key in actual.keys():
-        if key in desired.keys() or not partial:
-            assert_allclose(actual[key], desired[key], atol=atol, rtol=rtol, err_msg=f"Failed for key {key}")
+        assert set(actual.keys()) == set(desired.keys())
```

Comment:
why did you drop `assertEqual` for this check?

Reply:
`assertEqual` cannot be used because this function is not inside a `unittest.TestCase` class.

Reply:
Makes sense, did not see that